subcommands/devices/list.go

```diff
@@ -14,29 +17,111 @@ import (
 
 var (
 	deviceNoShared      bool
-	deviceNoOwner       bool
 	deviceByTag         string
 	deviceByFactory     string
 	deviceByGroup       string
 	deviceInactiveHours int
 	deviceUuid          string
+	showColumns         []string
 )
 
+type column struct {
+	FieldName string
+	Formatter func(d *client.Device, fieldName string) string
+}
+
+func defFormatter(d *client.Device, fieldName string) string {
+	e := reflect.ValueOf(d).Elem()
+	val := e.FieldByName(fieldName).Interface()
+	if reflect.TypeOf(val).String() == "[]string" {
```

Comment:
It's better to do this via a type assertion: `if str, ok := val.([]string); ok {`; reflect module is "unsafe" in Golang and should be the last resort.